pkg/deployer/lib/controller.go

```diff
@@ -154,8 +154,15 @@ func (c *controller) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		return reconcile.Result{}, err
 	}
 
-	var target *lsv1alpha1.Target
+	target, responsible, err := c.checkTargetResponsibility(ctx, logger, deployItem)
```

Comment:
The check of the target is now done twice. First in the new method `checkTargetResponsibility`, and then a second time after the method was called.

Reply:
good catch